target_chains/sui/contracts/sources/merkle_tree.move

```diff
@@ -0,0 +1,276 @@
+// Implementation of a Merkle tree in Move. Supports constructing a new tree
+// with a given depth, as well as proving that a leaf node belongs to the tree.
+module pyth::merkle {
+    use std::vector::{Self};
+    use sui::hash::{keccak256};
+    use wormhole::bytes::{Self};
+    use wormhole::bytes20::{Self, Bytes20, data};
+    use wormhole::cursor::{Self, Cursor};
+    use pyth::deserialize::{Self};
+
+    const MERKLE_LEAF_PREFIX: u8 = 0;
+    const MERKLE_NODE_PREFIX: u8 = 1;
+    const MERKLE_EMPTY_LEAF_PREFIX: u8 = 2;
+
+    const E_DEPTH_NOT_LARGE_ENOUGH_FOR_MESSAGES: u64 = 1212121;
+
+    // take keccak256 of input data, then return 20 rightmost bytes of result
+    fun hash(bytes: &vector<u8>): Bytes20 {
```

Comment:
not 100% sure if this is what ETH does. Pretty sure it takes the rightmost 20 bytes, but will have to confirm in the testing phase.

Reply:
I think it is the first 20 bytes. [This](https://github.com/pyth-network/pyth-crosschain/blob/main/pythnet/pythnet_sdk/src/hashers/keccak256_160.rs) is the hash we use on pythnet validator.